SharedSources/VideoViewController.swift

```diff
@@ -0,0 +1,124 @@
+//
+//  VideoViewController.swift
+//  VLC-iOS
+//
+//  Created by Carola Nitz on 11/28/17.
+//  Copyright © 2017 VideoLAN. All rights reserved.
+//
+
+import Foundation
+
+
+public protocol VLCVideoControllerDelegate: class {
+    func videoViewControllerDidSelectMediaObject(VLCVideoViewController: VLCVideoViewController, mediaObject:NSManagedObject)
+}
+
+public class VLCVideoViewController: UICollectionViewController, UICollectionViewDelegateFlowLayout
+{
+    private var mediaDataSource: VLCMediaDataSource
+    private let cellPadding:CGFloat = 5.0
+    public weak var delegate: VLCVideoControllerDelegate?
+
+    @available(iOS 11.0, *)
+    lazy var dragAndDropManager:VLCDragAndDropManager = {
+        let dragAndDropManager = VLCDragAndDropManager()
+        dragAndDropManager.delegate = mediaDataSource
+        return dragAndDropManager
+    }()
+
+    public convenience init(mediaDataSource:VLCMediaDataSource) {
+        self.init(collectionViewLayout: UICollectionViewFlowLayout())
+        self.mediaDataSource = mediaDataSource
+    }
+
+    public override init(collectionViewLayout layout: UICollectionViewLayout) {
+        self.mediaDataSource = VLCMediaDataSource()
+        super.init(collectionViewLayout: layout)
+    }
+
+    @available(*, unavailable)
+    required public init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override public func viewDidLoad() {
+
+        setupCollectionView()
+        setupNavigationbar()
+
+        _ = (MLMediaLibrary.sharedMediaLibrary() as AnyObject).perform(#selector(MLMediaLibrary.libraryDidAppear))
```

Comment:
this is a hack for now actually and needs to be replaced when we have the new medialibrary. it's just so that content is at least displayed here and you need to do the + 2.0 I should add a Todo and a ticket that this needs to be resolved properly